core/src/main/java/io/github/belgif/rest/guide/validator/core/ApiFunctions.java

```diff
@@ -109,10 +128,17 @@ private static Set<SchemaDefinition> getSubSchemas(SchemaDefinition schemaDefini
     }
 
     private static Set<SchemaDefinition> getRecursiveSubSchemas(SchemaDefinition schemaDefinition, Parser.ParserResult result, boolean includeTopLevelSchemas) {
+        return getRecursiveSubSchemas(schemaDefinition, result, includeTopLevelSchemas, new HashSet<>());
+    }
+
+    private static Set<SchemaDefinition> getRecursiveSubSchemas(SchemaDefinition schemaDefinition, Parser.ParserResult result, boolean includeTopLevelSchemas, Set<SchemaDefinition> checkedSubSchemas) {
+        checkedSubSchemas.add(schemaDefinition);
```

Comment:
A single structure for "checkedSubSchemas" and "output" could be used for this method:

`void addRecursiveSubSchemas(SchemaDefinition schemaDefinition, Parser.ParserResult result, boolean includeTopLevelSchemas, Set<SchemaDefinition> subSchemas)`